src/Foundatio.Parsers.SqlQueries/Extensions/SqlNodeExtensions.cs

```diff
@@ -0,0 +1,297 @@
+            {
+                context.AddValidationError("Field or DefaultFields is required for term queries.");
+                return String.Empty;
+            }
+
+            for (int index = 0; index < context.DefaultFields.Length; index++)
+            {
+                builder.Append(index == 0 ? "(" : " OR ");
+
+                var defaultField = GetFieldInfo(context.Fields, context.DefaultFields[index]);
+                if (defaultField.IsCollection)
+                {
+                    var dotIndex = defaultField.Field.LastIndexOf('.');
+                    var collectionField = defaultField.Field.Substring(0, dotIndex);
+                    var fieldName = defaultField.Field.Substring(dotIndex + 1);
+
+                    builder.Append(collectionField);
+                    builder.Append(".Any(");
+                    builder.Append(fieldName);
+                    builder.Append(".Contains(\"").Append(node.Term).Append("\")");
+                    builder.Append(")");
+                }
+                else
+                {
+                    builder.Append(defaultField.Field).Append(".Contains(\"").Append(node.Term).Append("\")");
+                }
+
+                if (index == context.DefaultFields.Length - 1)
+                    builder.Append(")");
+            }
+
+            return builder.ToString();
+        }
+
+        // support overriding the generated query
+        if (node.TryGetQuery(out string query))
+            return query;
+
+        var field = GetFieldInfo(context.Fields, node.Field);
+
+        if (node.IsNegated.HasValue && node.IsNegated.Value)
+            builder.Append("NOT ");
+
+        if (field.IsCollection)
+        {
+            var index = node.Field.LastIndexOf('.');
+            var collectionField = node.Field.Substring(0, index);
+            var fieldName = node.Field.Substring(index + 1);
+
+            builder.Append(collectionField);
+            builder.Append(".Any(");
+            builder.Append(fieldName);
+
+            if (node.IsNegated.HasValue && node.IsNegated.Value)
+                builder.Append(" != ");
+            else
+                builder.Append(" = ");
+
+            AppendField(builder, field, node.Term);
+
+            builder.Append(")");
+        }
+        else
+        {
+            builder.Append(node.Field);
+            if (node.IsNegated.HasValue && node.IsNegated.Value)
+                builder.Append(" != ");
+            else
+                builder.Append(" = ");
+
+            AppendField(builder, field, node.Term);
+        }
+
+        return builder.ToString();
+    }
+
+    private static void AppendField(StringBuilder builder, EntityFieldInfo field, string term)
+    {
+        if (field == null)
+            return;
+
+        if (field.IsNumber || field.IsBoolean)
+            builder.Append(term);
+        else if (field is { IsDate: true })
+            builder.Append("DateTime.Parse(\"" + term + "\")");
```

Comment:
I question the name ToSqlString, when this isn't straight sql but a EF/Linq DSL.

Reply:
I changed it to ToDynamicLinqString